source/filesystem.hpp

```diff
@@ -23,7 +23,7 @@ namespace reshade::filesystem
 	public:
 		path() { }
 		path(const char *data) : _data(data) { }
-		path(const std::string &data) : _data(data) { }
+		path(std::string data) : _data(std::move(data)) { }
```

Comment:
The latter is move semantics friendly. Let's compare three cases:

1.
```c++
std::string string = "FOO";
path myPath(string);
```
Before the change, `path` constructor does one copy (from `data` to `_data`). After the change, `path` constructor does one copy (to `data`) and one move (from `data` to `_data`). Move constructing overhead is most likely negligible, plus it's not unlikely the compiler will notice it and optimize the move out.

2.
```c+++
std::string string = "FOO";
path myPath(std::move(string));
```

Before the change, there is no constructor which takes an rvalue so it degrades to lvalue and becomes identical to case 1. - thus doing a copy. After the change, `data` is move constructed and so is `_data` - so we get to use move semantics properly.

3.
```c++
path myPath(std::string("FOO"));
```

Before the change, the temporary is constructed, copied (from `data` to `_data`) and destructed. After the change, the temporary is constructed and moved.

Of course, another way to solve it is to just add another constructor taking `std::string&&` instead of trying to unify the two. What I learnt recently though is that it's perfectly reasonable to pass arguments by value instead of const reference when the constructor is supposed to copy them. It's especially useful if you have multiple arguments and you'd like to accept them as either lvalue or rvalue - instead of resorting to hacks or creating overloads for every possible combination of parameters, you just pass them by value and move to class fields.